### PR TITLE
Allow the same user to have multiple top level comments

### PR DESCRIPTION
**Pull request by Soulofdestiny**
still takes the highest number from all of those
this bug was found by /u/superhammerboy

### CODE REVIEW

File: geostackr.py [hunk 1]
```diff
@@ -165,7 +165,11 @@ def get_score_list(submission, series_config: Dict[str, UserScores]) -> Dict[str
             if comment.author.name not in IGNORE_USERS | series_config['ignore']:
                 number = get_goal_number_from_text(series_config, comment.body)
                 if number:
-                    score_list[comment.author.name] = number
+                    # Check if there are more top_level_comments from the same user which contains numbers and take the highest
+                    if comment.author.name in score_list:
+                        score_list[comment.author.name] = number if score_list[comment.author.name] < number else score_list[comment.author.name]
+                    else:
+                        score_list[comment.author.name] = number
```
Comment (LiquidFun) [inline geostackr.py]: Good idea! Forgot to handle this.

One issue though is that depending on the `goal` (either `highest` or `lowest`) for the series this doesn't have to be the largest number. Sometimes the smallest one may be prioritized. Then this `score_list[comment.author.name] < number` would be incorrect.

So you could do something like this:

```python
goal_function = get_goal_function(series_config)
score_list[comment.author.name] = goal_function(number, score_list.get(comment.author.name, number))
```

Explanation: .get on a dict returns the key, or if it doesn't exist the second value. In this case `number`. So if the `goal_function` is `max` we get either `max(number, number)` or `max(number, score_list[comment.author.name])`.

**Review by LiquidFun — CHANGES_REQUESTED** [review]
Good idea! Doesn't work yet though if `goal` is `lowest`.

File: geostackr.py (hunk 1 above)
Comment (LiquidFun) [reply geostackr.py]: Ah, okay, you already saw this, didn't see the your comment, sorry.

Comment (Soulofdestiny) [reply geostackr.py]: fixed

**Pull request merged**